### ISSUE TITLE
Bauh closes when settings button is clicked

### ISSUE THREAD

**Issue by Lennox84** [vinifmor/bauh#69]
Wanted to try giving the web apps a go so I went to open the settings and when I click it bauh closes. This has happened on this release and the previous. I thought I'd wait for an update to see if that solved it but it did not.

I'm on 0.8.3-1.
Linux T450 5.4.18-1-MANJARO #1 SMP PREEMPT Thu Feb 6 11:41:30 UTC 2020 x86_64 GNU/Linux

Happy to run/do any debugging needed, but you'll have to explain how :-)

**Comment by vinifmor** [comment]
Hello @Lennox84 , try the latest stable version ( **0.8.4** ) already available in AUR. If the issue still persists, please open bauh through a terminal ( ```bauh``` ) and paste here any trace left so we can investigate what's going on.

**Comment by masteor** [comment]
i have the same issue - my trace:
qt5ct: using qt5ct plugin
QPainter::begin: Paint device returned engine == 0, type: 2
Segmentation fault (core dumped)

**Comment by vinifmor** [comment]
@masteor , perhaps there is a package not installed on your system. Try the manual installation and see if the error happens as well.
```
python3 -m venv bauh_env
bauh_env/bin/pip install bauh
bauh_env/bin/bauh
```

**Comment by Lennox84** [comment]
I haven't installed the AUR version yet and Manjaro is still only on
0.8.3-1,
I'll install the AUR if I really need to but happy to wait if it doesn't
take too long for Manjaro to get updated.

just to add to the above/below. I ran an Strace and the bottom section of
results was: (I'm comfortable in using Linux but don't know how to debug or
detect when something break so wanted to give this a try. Please let me
know if this is helpful or what would be best to run to troubleshoot)

sendmsg(14, {msg_name={sa_family=AF_UNIX,
sun_path="/run/systemd/journal/socket"}, msg_namelen=30,
msg_iov=[{iov_base="MESSAGE", iov_len=7}, {iov_base="\n", iov_len=1},
{iov_base="\352\3\0\0\0\0\0\0", iov_len=8}, {iov_base="Traceback (most
recent call last"..., iov_len=1002}, {iov_base="\n", iov_len=1},
{iov_base="PRIORITY=1", iov_len=10}, {iov_base="\n", iov_len=1},
{iov_base="CODE_FUNC=unknown", iov_len=17}, {iov_base="\n", iov_len=1},
{iov_base="CODE_LINE=0", iov_len=11}, {iov_base="\n", iov_len=1},
{iov_base="CODE_FILE=unknown", iov_len=17}, {iov_base="\n", iov_len=1},
{iov_base="QT_CATEGORY=default", iov_len=19}, {iov_base="\n", iov_len=1},
{iov_base="SYSLOG_IDENTIFIER=", iov_len=18}, {iov_base="python3",
iov_len=7}, {iov_base="\n", iov_len=1}], msg_iovlen=18, msg_controllen=0,
msg_flags=0}, MSG_NOSIGNAL) = 1124
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1], [], 8) = 0
getpid()                                = 8950
gettid()                                = 8950
tgkill(8950, 8950, SIGABRT)             = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=8950, si_uid=1000}
---
+++ killed by SIGABRT (core dumped) +++

Cheers

**Comment by vinifmor** [comment]
@Lennox84 ,

What desktop environment are you currently using ? Also paste here the result of the following command: `python -c "import locale; print(locale.getdefaultlocale())"`

**Comment by Lennox84** [comment]
Hi Vin,

I'm running KDE - plasmashell 5.17.5

Output: ('en_AU', 'UTF-8')

Cheers

**Comment by vinifmor** [comment]
Hum... I use KDE at home, and had no issue so far. I'm going to try some scenarios to see if I can reproduce this issue.

Cheers

**Comment by vinifmor** [comment]
@Lennox84 , couldn't reproduce on my KDE end, but I added some error handling related to icons loading that might be related. Please, install**bauh-staging** from AUR ( 0.8.5.RC-3 ) and let me know if the issue is fixed. Consider launching bauh through the command line ( `bauh` ) to catch any error message.

**Comment by vinifmor** [comment]
@Lennox84 @masteor is this issue still happening on your ends ?

**Comment by Lennox84** [comment]
Yeah still happening for me, removed repo version and installed AUR. Still
happening.

I've just installed a fresh version of manjaro on another machine. Will
install bauh fresh on new machine in the next few hours and let you know
how it goes. I'll also update the original machine today and see if the
problem still exists.

Will update soon 😁

**Comment by vinifmor** [comment]
Thanks

**Comment by Lennox84** [comment]
Ok so fresh install of Manjaro with xfe went smoothly. Also installed it on
Pop OS with zero issues 😊 Love the new v9 upgrades. Starting to really
feel like a standalone package.

Great work Vin! Going to test on my other kde install (the original with
the issue). Will respond in the next few hours.

Thank you again Vin 💪

**Comment by Lennox84** [comment]
Ok so issue still exists on 8.5 and 9.0
The + items work, installed digiKam.Appimage 😇
Cleaned Cache and did a re-sync
Tried out system upgrade
All looks good 💪

Are there any tests you would like me to try? I'm wondering if there's a
setting I've changed in the past which is causing an error, is there a file
or log I can backup and replace to see if the trigger for the issue is in
there?? just a thought as it's so random.

Again thank you so much for your hard work, i love bauh and thinks ts great!

Cheers
Lennox

**Comment by vinifmor** [comment]
Hi @Lennox84 , thank you for the effort in testing the new features. The users feedback is very important for the future of bauh.

In relation to the config file... it is possible. Please, try to execute the following commands and tell me if the settings dialog shows up:
```
rm -rf ~/.config/bauh
bauh --settings --logs
```

**Comment by Lennox84** [comment]
WooHoo!!!

Success, it works, the only problem I thought of now is we should have
backed up the original and I could have sent it to you to review.

Again thanks for all of your hard work. It's working great on both my
Manjaro machines 1xKDE & 1xXFCE.
I haven't needed to use pamac for a few days now, been solely using bauh
and it's working great.

Kind regards,
Lennox84

**Comment by vinifmor** [comment]
Great @Lennox84 ! Well, it seems to be an issue while parsing the config files. I'm going to add a treatment to prevent this sort of error. I'm going to publish the first minor release (0.9.1) in the next days (it fixes some behaviors related to Arch support). Some of the fixes are already published in **bauh-staging** if you want to give them a shot.

**Comment by tom-tullio** [comment]
Just another data point -- I was experiencing the same issue on 5.6.8-1-MANJARO/Cinnamon with **0.9.2.RC-4**

Clicking the settings gear closes the application.  Terminal showed:

`Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/bauh/view/qt/window.py", line 1228, in show_settings
    self.settings_window = SettingsWindow(self.manager, self.i18n, self.screen_size, self)
  File "/usr/lib/python3.8/site-packages/bauh/view/qt/settings.py", line 29, in __init__
    self.settings_model = self.manager.get_settings(screen_size.width(), screen_size.height())
  File "/usr/lib/python3.8/site-packages/bauh/view/core/controller.py", line 422, in get_settings
    return self.settings_manager.get_settings(screen_width=screen_width, screen_height=screen_height)
  File "/usr/lib/python3.8/site-packages/bauh/view/core/settings.py", line 69, in get_settings
    tabs.append(self._gen_ui_settings(core_config, screen_width, screen_height))
  File "/usr/lib/python3.8/site-packages/bauh/view/core/settings.py", line 171, in _gen_ui_settings
    default_option=[o for o in style_opts if o.value == cur_style][0],
IndexError: list index out of range
Aborted (core dumped)
`

Removing config files didn't help, but launching with `bauh --settings --logs` did.  All is well now.

**Comment by vinifmor** [comment]
@tom-tullio it seems a Qt Style previously set was not available on the system anymore. When you reset bauh it started to use the default theme (Fusion) and could render normally. I'm going to implement a workaround for this scenario. Thank you for reporting the issue.

**Comment by vinifmor** [comment]
I've just published a new RC to **bauh-staging** (**0.9.2.RC-5**) fixing this style issue. The final release should be available soon. Any problem, let me know

**Comment by vinifmor** [comment]
The final release with this fix is out. If this problem shows up again, please, reopen the issue.

Cheers
